### PR TITLE
IntSet

### PR DESCRIPTION
**Pull request by nofun97**
Basic IntSet with some fixed bugs, iterator, and small tests

### CODE REVIEW

**Comment by nofun97** [comment]
After looking through the tests, I should probably use higher numbers.

File: intset_test.go [hunk 1]
```diff
@@ -0,0 +1,212 @@
...
+	assert.True(t, mappedSet.EqualSet(fullSet.Map(subtract)))
+	assert.True(t, NewIntSet().EqualSet(NewIntSet().Map(subtract)))
+}
+
+func generateIntArrayAndSet() ([]int, IntSet) {
+	arr := make([]int, 0, maxIntArrLen)
+	curr := float64(1.0)
+	for i := 1; i < maxIntArrLen; i++ {
+		arr = append(arr, int(curr))
+		curr *= math.Pow(2, 64/math.Pow(10, 6))
```
Comment (marcelocantos) [inline intset_test.go]: The multiplier is a constant. Calculate it once outside the function.

Comment (marcelocantos) [inline intset_test.go]: math.Pow(10, 6) = 1e6

File: intset_benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+import (
+	"testing"
+)
+
+func benchmarkNewIntSet(b *testing.B, n int) {
+	arr, _ := generateIntArrayAndSet()
+	max := n
+	if n > len(arr) {
+			b.Logf("n of %d is bigger than the generated array length of %d", n, len(arr))
```
Comment (marcelocantos) [inline intset_benchmark_test.go]: Make it so this doesn't happen.

**Review by marcelocantos — CHANGES_REQUESTED** [review]

File: intset_test.go (hunk 1 above)
Comment (nofun97) [reply intset_test.go]: added

Comment (nofun97) [reply intset_test.go]: change added

File: intset_benchmark_test.go (hunk 2 above)
Comment (nofun97) [reply intset_benchmark_test.go]: not it generates n numbers

Comment (marcelocantos) [reply intset_benchmark_test.go]: ??

File: intset_benchmark_test.go [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+func BenchmarkNewIntSet100(b *testing.B) {benchmarkNewIntSet(b, 100)}
+func BenchmarkNewIntSet1k(b *testing.B) {benchmarkNewIntSet(b, 1000)}
+func BenchmarkNewIntSet100k(b *testing.B) {benchmarkNewIntSet(b, 100000)}
+func BenchmarkNewIntSet1M(b *testing.B) {benchmarkNewIntSet(b, 1000000)}
+
+func BenchmarkWithIntSet100(b *testing.B) {benchmarkWithIntSet(b, 100)}
+func BenchmarkWithIntSet1k(b *testing.B) {benchmarkWithIntSet(b, 1000)}
+func BenchmarkWithIntSet100k(b *testing.B) {benchmarkWithIntSet(b, 100000)}
+func BenchmarkWithIntSet1M(b *testing.B) {benchmarkWithIntSet(b, 1000000)}
```
Comment (marcelocantos) [inline intset_benchmark_test.go]: Missing \n

**Review by marcelocantos — CHANGES_REQUESTED** [review]

File: intset_benchmark_test.go (hunk 2 above)
Comment (nofun97) [reply intset_benchmark_test.go]: *now

File: intset_benchmark_test.go (hunk 3 above)
Comment (nofun97) [reply intset_benchmark_test.go]: whoops, newline added.

**Comment by nofun97** [comment]
In addition to the base features, now it has some basic operations, including, `Intersection`, `Union`, `EqualSet`, `Where`, and `Map`.

**Review by marcelocantos — APPROVED** [review]

**Pull request merged**